### PR TITLE
fix(shipping): CHECKOUT-7627 Google AddressSelector sometimes autofills the wrong city for Canada addresses

### PR DESCRIPTION
**Pull request by animesh1987**
## What?
Fix issue where Google maps API sometimes fills the incorrect value in the city field.

## Why?
This was raised in an issue here https://github.com/bigcommerce/checkout-js/issues/1390.

## Testing / Proof
- circle
- screenshot
Screenshot 2023-08-04 at 10 29 20 am
Screenshot 2023-08-04 at 10 29 52 am

@bigcommerce/checkout

### CODE REVIEW

File: packages/core/src/app/address/googleAutocomplete/AddressSelectorCA.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import AddressSelector from './AddressSelector';
+
+export default class AddressSelectorUK extends AddressSelector {
```
Comment (bc-peng) [inline packages/core/src/app/address/googleAutocomplete/AddressSelectorCA.ts:3]:
```suggestion
export default class AddressSelectorCA extends AddressSelector {
```

Comment (bc-peng) [reply packages/core/src/app/address/googleAutocomplete/AddressSelectorCA.ts:3]: Now you see why I don't like using `default` 🥴

File: packages/core/src/app/address/googleAutocomplete/AddressSelectorCA.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    getCity(): string {
+        return this._get('sublocality_level_1', 'long_name') || this._get('locality', 'long_name');
```
Comment (bc-peng) [inline packages/core/src/app/address/googleAutocomplete/AddressSelectorCA.ts:5]: Do we need someone to verify this rule is valid for all CA addresses?

File: packages/core/src/app/address/googleAutocomplete/AddressSelectorCA.ts (hunk 1 above)
Comment (animesh1987) [reply packages/core/src/app/address/googleAutocomplete/AddressSelectorCA.ts:3]: Wow that's next level copy paste error 🤦

**Review by bc-peng — APPROVED** [review]

**Pull request merged**